### PR TITLE
Greedy roles

### PR DESCRIPTION
**Pull request by zepthro**
Simple fix for #155.

I looked into Greedy[str], but you cannot use a Greedy[str] type, so I did the next best thing, I used split(' ').

Here's a few examples of the changes
Old functionality:
image
New functionality:
image

### CODE REVIEW

File: bot/cogs/assignable_roles_cog.py [hunk 1]
```diff
@@ -23,21 +23,21 @@ def __init__(self, bot) -> None:
     @ext.long_help('Lists all roles that have been marked as assignable in this server')
     @ext.short_help('Defines custom assignable roles')
     @ext.example('roles')
-    async def roles(self, ctx, *, input_role: str = None) -> None:
-        if input_role is None:
+    async def roles(self, ctx, *, input_roles: str = None) -> None:
```
Comment (Jay-Madden) [inline bot/cogs/assignable_roles_cog.py:26]: this needs to be a `greedy[discord.Roles]` converter

File: bot/cogs/assignable_roles_cog.py [hunk 2]
```diff
@@ -23,21 +23,21 @@ def __init__(self, bot) -> None:
...
-                raise BadArgument
+                if not await self.check_role_assignable(ctx, input_role):
+                    raise BadArgument
 
-            await self.set_role(ctx, role)
+                await self.set_role(ctx, role)
         
-        except BadArgument:  # If RoleConverter failed
-            await self.find_possible_roles(ctx, input_role)
+            except BadArgument:  # If RoleConverter failed
```
Comment (Jay-Madden) [inline bot/cogs/assignable_roles_cog.py:39]: we will need to handle this differently for multi roles

File: bot/cogs/assignable_roles_cog.py [hunk 3]
```diff
@@ -23,21 +23,19 @@ def __init__(self, bot) -> None:
     @ext.long_help('Lists all roles that have been marked as assignable in this server')
     @ext.short_help('Defines custom assignable roles')
     @ext.example('roles')
-    async def roles(self, ctx, *, input_role: str = None) -> None:
-        if input_role is None:
+    async def roles(self, ctx, input_roles: commands.Greedy[discord.Role] = None, bad_args: str = None) -> None:
+        if input_roles is None or bad_args:
```
Comment (Jay-Madden) [inline bot/cogs/assignable_roles_cog.py:27]: the check for bad args needs to come at the end, you can have a valid role in between invalid ones

Comment (Jay-Madden) [reply bot/cogs/assignable_roles_cog.py:27]: make the default for input_roles an empty list then you can check bad args at the end

**Comment by Jay-Madden** [comment]
hmm, how does this handle roles with spaces. For example Special Topics

**Review by Jay-Madden — CHANGES_REQUESTED** [review]
see above question

**Comment by Jay-Madden** [comment]
Im not convinced of the possibility of this pr in a clean and concise way, what are your thoughts?

**Comment by zepthro** [comment]
> Im not convinced of the possibility of this pr in a clean and concise way, what are your thoughts?

I agree. I think we should make a comment on #155, but not close the issue. If someone comes along and figures out an elegant solution, then great.
Additionally, we can close this pull. I don't have a solution as of now.

**Pull request closed without merging**